vault/password_policy.go

```diff
@@ -0,0 +1,125 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func readPasswordPolicy(client *api.Client, name string) (map[string]interface{}, error) {
+	r := client.NewRequest("GET", fmt.Sprintf("/v1/sys/policies/password/%s", name))
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	resp, err := client.RawRequestWithContext(ctx, r)
+	if resp != nil {
+		defer resp.Body.Close()
+		if resp.StatusCode == 404 {
+			return nil, nil
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	secret, err := api.ParseSecret(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if secret == nil || secret.Data == nil {
+		return nil, errors.New("data from server response is empty")
+	}
+	return secret.Data, nil
+}
+
+// PutPasswordPolicy creates a password policy from a given name and policy string
+func PutPasswordPolicy(client *api.Client, name string, body map[string]interface{}) error {
+	r := client.NewRequest("PUT", fmt.Sprintf("/v1/sys/policies/password/%s", name))
+	if err := r.SetJSONBody(body); err != nil {
+		return err
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	resp, err := client.RawRequestWithContext(ctx, r)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+// DeletePasswordPolicy deletes a given password policy
+func DeletePasswordPolicy(client *api.Client, name string) error {
+	r := client.NewRequest("DELETE", fmt.Sprintf("/v1/sys/policies/password/%s", name))
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	resp, err := client.RawRequestWithContext(ctx, r)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+	return err
+}
+
+func passwordPolicyDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	name := d.Id()
+
+	log.Printf("[DEBUG] Deleting %s password policy from Vault", name)
+
+	err := DeletePasswordPolicy(client, name)
```

Comment:
I only see this used in one place, so it might make sense to just add this logic to `passwordPolicyWrite`. Thoughts?